File: LeetCode/Basic/007/main.go
package main

func reverse(x int) int {
	var num, reversed int
	var isNegative bool = false
	if x < 0 {
		isNegative = true
		num = -x
	} else {
		num = x
	}
	for num != 0 {
		reversed = reversed*10 + num%10
		num /= 10
	}
	if isNegative {
		return (-1) * reversed
	}
	return reversed
}

File: LeetCode/Basic/007/main.py
class Solution:
    def reverse(self, x: int) -> int:
        num, reversed = 0, 0
        is_negative = False

        if x < 0:
            is_negative = True

        num = abs(x)
        while num != 0:
            reversed = reversed*10+num % 10
            num /= 10

        if is_negative:
            return (-1)*reversed

        return reversed
